js/cpu6502.ts

```diff
@@ -1158,12 +1158,26 @@ export default class CPU6502 {
             off = b;
         } else {
             page = a >> 8;
-            off = page & 0xff;
+            off = a & 0xff;
         }
         return this.memPages[page].read(page, off);
     }
 
-    public write(page: byte, off: byte, val: byte) {
+    public write(addr: word, val: byte): void;
+    public write(page: byte, off: byte, val: byte): void;
+
+    public write(a: number, b: number, c?: byte): void {
```

Comment:
I wondering if there is some fancy destructuring that could be done. I don't know if it's worth it, but maybe:
```
let [page, off, val] = c !== undefined ? [a, b, c] : [a >> 8, a & 0xff, b];
```